contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/insertion/DefaultInsertionCostCalculator.java

```diff
@@ -55,6 +60,23 @@ public double calculate(DrtRequest drtRequest, Insertion insertion, DetourTimeIn
 			return INFEASIBLE_SOLUTION_COST;
 		}
 
+		// divert right now
+		if(insertion.pickup.index == 0) {
+			Waypoint nextWaypoint = insertion.pickup.nextWaypoint;
+			// there is an existing stop following the new insertion
+			if(nextWaypoint instanceof Waypoint.Stop && nextWaypoint != insertion.pickup.newWaypoint) {
+				// passengers are being dropped off == may be close to arrival
+				if(!((Waypoint.Stop) nextWaypoint).task.getDropoffRequests().isEmpty()) {
+					double nextArrival = nextWaypoint.getArrivalTime();
+					double departureTime = insertion.vehicleEntry.start.getDepartureTime();
+					//arrival is very soon
+					if (nextArrival - departureTime < drtConfigGroup.fixedApproachTime) {
+						return INFEASIBLE_SOLUTION_COST;
+					}
+				}
+			}
+		}
+
```

Comment:
Have you maybe thought about wrapping `DefaultInsertionCostCalculator` with `FixedArrivalTimeCostCalculator`. This way we can use guice to add this additional condition while keeping the config untouched?

Reply:
Yes I thought about this but I liked the fact of having it configurable through the config. If you think it is too specific we can also go with the wrapper option - but that means one would need to always set it programmatically right?

Reply:
I think this is a nice feature that can be potentially used in many scenarios, so having it in the config sounds good to me.

I am not sure if all algorithms will be able to support this feature, but that is a different problem that we need to address in "drt config v2.0"...

Reply:
👍